homeassistant/components/light/osramlightify.py

```diff
@@ -197,22 +236,71 @@ def turn_off(self, **kwargs):
             _LOGGER.debug("turn_off requested transition time for light:"
                           " %s is: %s ",
                           self._name, transition)
-            self._light.set_luminance(0, transition)
+            self._luminary.set_luminance(0, transition)
         else:
             transition = 0
             _LOGGER.debug("turn_off requested transition time for light:"
                           " %s is: %s ",
                           self._name, transition)
-            self._light.set_onoff(0)
-            self._state = self._light.on()
-
+        self._luminary.set_onoff(0)
+        self._state = False
         self.schedule_update_ha_state()
```

Comment:
Can you remove this line? it's not needed since should_poll is defaulting to True and so this is automatically called.